darwin/DarwinProcess.c

```diff
@@ -240,6 +240,12 @@ void DarwinProcess_setFromKInfoProc(Process* proc, const struct kinfo_proc* ps,
    proc->updated = true;
 }
 
+double ticks_to_nanoseconds(const double ticks) {
+   const double secs_per_tick = 1.0 / (double)sysconf(_SC_CLK_TCK);
```

Comment:
Probably best to cache the results of this sysconf call so that we're not recalculating it on every sample?  (perhaps a global 'hertz' var set from either the darwin Platform_init or ProcessList_new routine)

sysconf can also fail, so doing this call once on startup will make error checking / handling easier.

Reply:
Agreed! Though I'm not a fan of `herz`, I'd rather something like `clock_ticks_per_s`. Thoughts?

I just feeling out if this PR is a move in the right direction.

I noticed that a *lot* of this code-base is unclear about units, exact derivations, etc., so I'm not sure if this kind of change would be desired or not.

Reply:
> Agreed! Though I'm not a fan of `herz`, I'd rather something like `clock_ticks_per_s`. Thoughts?

Sure, doesn't have to be hertz - if its multi-words like that though camel case is more commonly used in htop (see darwin/Platform.c for some examples).

> I just feeling out if this PR is a move in the right direction.

Looks like its moving in a less broken direction - so two thumbs up from me.

Reply:
Ah yes, I didn't notice. Looks like local vars are snake case, and functions/types are camel case. I'll change my identifiers to match.

Some other questions:

1. Should I add an explicit but technically redundant `#include <unistd.h>`? It's what defines `sysconf`, but it's also already indirectly included (probably by CoreFoundation, but I haven't checked.

2. The return type of `sysconf` is `long`. Should the cache variable match that and be a `long`, or use `uint64_t` like the rest? Or should I cache it after conversion to `double`?

3. I saw that `sysconf(_SC_CLK_TCK)` is also called in `LinuxProcess_adjustTime`. They called the cache var `jiffy`. It appears that they use a lazy-initialization pattern, that makes `sysconf(_SC_CLK_TCK)` called upon the first call to `LinuxProcess_adjustTime`.

    They fail silently if the `errno` is set, by just assuming 100 Hz. That seems... like the wrong choice to me. Wouldn't it be better to fail loudly than to assume a potentially wrong value? I feel like I'm pulling a thread here haha.

Reply:
>     1. Should I add an explicit but technically redundant `#include <unistd.h>`? It's what defines `sysconf`, but it's also already indirectly included (probably by CoreFoundation, but I haven't checked.

I'd not add it explicitly unless its needed to compile - as you say, chances are its included indirectly already.

>     2. The return type of `sysconf` is `long`. Should the cache variable match that and be a `long`, or use `uint64_t` like the rest? Or should I cache it after conversion to `double`?

I recommend you do as much work up front as possible, at startup, and as little work as possible during each sample.
Thus, I think its best to store it as a double (like you had it in the original patch).

>     3. I saw that `sysconf(_SC_CLK_TCK)` is also called in `LinuxProcess_adjustTime`. They called the cache var `jiffy`. It appears that they use a lazy-initialization pattern, that makes `sysconf(_SC_CLK_TCK)` called upon the first call to `LinuxProcess_adjustTime`.

There's a good chance this code can be improved also.

>        They fail silently if the `errno` is set, by just assuming 100 Hz. That seems... like the wrong choice to me. Wouldn't it be better to fail loudly than to assume a potentially wrong value? I feel like I'm pulling a thread here haha.

Heh - yep.  It definitely would be better to fail htop at startup if this call fails - on Linux too, IMO.

Reply:
> I recommend you do as much work up front as possible, at startup, and as little work as possible during each sample.
Thus, I think its best to store it as a double (like you had it in the original patch).

I had a change of mind on this. Though `double` suits my usecase, some other feature might end up re-using it and converting it to `uint64_t`. It's just one `cvttsd2si` instruction anyway, so I was leaning towards keeping it as its original `long`.


>Heh - yep. It definitely would be better to fail htop at startup if this call fails - on Linux too, IMO.

Opened #384 to track this.

Reply:
@natoscott NVM, the style guide suggests to add such headers: https://github.com/htop-dev/htop/blob/master/docs/styleguide.md#import-and-use-of-headers